PB-118 Advanced sharing

aboredvaro opened the pull request.

CandelR committed:
Removed duplicated code

aboredvaro committed:
Advanced Sharing Dialog

aboredvaro committed:
Add onRemove user method

aboredvaro committed:
[FE] Owner can stop sharing an item

aboredvaro committed:
Added translations for ES, FR, CN, IT and RU

aboredvaro committed:
Hide changes in production

- Change requested by Sergio

CandelR committed:
Merged with refactored drive context menu and added manage acces and share menu items - WIP

CandelR committed:
Fix open popover with one click and avatar flicker

CandelR committed:
Merge branch 'feature/PB-118-advanced-sharing' into feature/PB-618-owner-chan-share-file-or-folder

CandelR committed:
Added logic to show new Share dialog in drive, recents and shared links views

aboredvaro committed:
Add access request screen and methods

CandelR committed:
Removed unnecessary condition in ShareDialog

CandelR committed:
Merge branch 'master' into feature/PB-118-advanced-sharing

CandelR committed:
Extract components from inside ShareDialog to avoid destroy in every render

rafijv committed:
[PB-672] create dialog allow owner share files with users

CandelR commented on src/app/drive/components/DriveItemDropdownActions/DriveItemDropdownActions.tsx:
```diff
@@ -99,31 +103,37 @@ const FileDropdownActions = (props: FileDropdownActionsProps) => {
+          <Users className="mr-1" size={20} />
+          <span>{translate('drive.dropdown.share')}</span>
+        </Dropdown.Item>
+      ) : null}
```
We have to hide this in production as the ShareDialg, isn't it?

CandelR commented on src/app/shared/components/List/index.tsx:
```diff
@@ -164,26 +164,50 @@ ListProps<T, F>): JSX.Element {
+      if (!disableKeyboardShortcuts) handleBackspaceKeyPressed();
+    },
+    [selectedItems, disableKeyboardShortcuts],
+  );
```
Maybe we could simplify the use of `!disabledKeyboardShortcuts` conditional in this way:
```
const handleKeyPress = (shortcutKey, action) => {
  return () => {
    if (!disableKeyboardShortcuts) action();
  };
};

useHotkeys('esc', handleKeyPress('esc', unselectAllItems), [selectedItems, disableKeyboardShortcuts]);
useHotkeys('enter', handleKeyPress('enter', executeClickOnSelectedItem), [selectedItems, disableKeyboardShortcuts]);
useHotkeys('r', handleKeyPress('r', handleRKeyPressed), [selectedItems, disableKeyboardShortcuts]);
useHotkeys('backspace', handleKeyPress('backspace', handleBackspaceKeyPressed), [selectedItems, disableKeyboardShortcuts]);
```

rafijv committed:
merge branch feature/PB-118-advanced-sharing into feat/PB-672-owner-can-share-files-with-users-by-invitation

rafijv committed:
[PB-672] add some fixes and translations

aboredvaro committed:
Added changes requested

- Hide "Share" action button on production
- Handle keyboard shortcuts depending on keyboard shortcut disabled state

rafijv committed:
[PB-672] remove comments and fix styles

CandelR committed:
Added translations

CandelR committed:
Merge pull request #724 from internxt/feat/PB-672-owner-can-share-files-with-users-by-invitation

[PB-672] feature/Owner can share files with users by invitation

rafijv committed:
[PB-665] add request access dialog

rafijv committed:
[PB-665] hide RequestAccesDialog

CandelR committed:
Added animation on deny user request

rafijv committed:
[PB-665] remove dialog and add view to request access

rafijv committed:
[PB-665] fix some issues

rafijv committed:
[PB-665] fix magic number

CandelR committed:
Merge pull request #725 from internxt/feat/PB-665-user-can-request-access

[PB-665]: feat/user can request access to a file or folder

CandelR committed:
Merge branch 'master' into feature/PB-618-owner-chan-share-file-or-folder

CandelR committed:
Added production condition to not show advanced sharing UI

CandelR committed:
Merge branch 'feature/PB-118-advanced-sharing' into feature/PB-618-owner-chan-share-file-or-folder

CandelR committed:
Merge pull request #730 from internxt/feature/PB-618-owner-chan-share-file-or-folder

[PB-618]: feature/ Owner can manage access requests from the "Manage access" dialog

CandelR committed:
Added node env production to ci .env

aboredvaro committed:
Fixed spacing

aboredvaro commented on src/app/drive/components/ShareInviteDialog/ShareInviteDialog.tsx:
```diff
@@ -0,0 +1,231 @@
+
+      <div className="p-5">
+        <form className="m flex w-full" onSubmit={handleSubmit(onAddInviteUser)}>
+          <input
```
1. Input used here should be:
`import Input from 'app/shared/components/Input';`

2. If the user types a comma "," it should have the same effect as pressing enter (add it to the "Users to invite" list and empty the input)

rafijv replied:
Done

aboredvaro commented on src/app/drive/components/ShareInviteDialog/ShareInviteDialog.tsx:
```diff
@@ -0,0 +1,231 @@
+                    <span className="capitalize">{userRole}</span>
+                    <CaretDown
+                      size={24}
+                      className={`${open ? 'rotate-180 transform transition' : 'rotate-0 transform transition'}`}
```
There should be no rotation animations in the dropdowns (here and in the user list)

rafijv replied:
Done

aboredvaro commented on src/app/share/views/ShareView/ShareFileView.tsx:
```diff
@@ -270,7 +270,7 @@ export default function ShareFileView(props: ShareViewProps): JSX.Element {
                 {getFormatFileName()}
               </abbr>
-              <span className="text-cool-gray-60">{getFormatFileSize()}</span>
+              <span className="text-cool-gray-60">{getFormatFileSize()}lala</span>
```
Remove "lala"

CandelR replied:
I mark it as solved because I saw it in another [task](https://github.com/internxt/drive-web/pull/732) that has to be merged and it is already removed.

aboredvaro commented on src/app/share/views/ShareView/ShareFileView.tsx:
```diff
@@ -286,6 +286,7 @@ export default function ShareFileView(props: ShareViewProps): JSX.Element {
                         font-medium text-blue-60 active:bg-blue-20 active:bg-opacity-65"
             >
               <UilEye height="20" width="20" />
+              lala
```
Remove "lala"

CandelR replied:
I mark it as solved because I saw it in another [task](https://github.com/internxt/drive-web/pull/732) that has to be merged and it is already removed.

CandelR committed:
Merge branch 'feature/PB-118-advanced-sharing' into PB-666-manage-access-requests

Commit:
Modified ShareInviteDialog in order to fit the ShareDialog structure

Commit:
Added status to every request and animation when is accepted

Commit:
Removed text added by mistake

Commit:
Some style changes

rafijv committed:
[PB-672] add Input component and submit form with comma

Commit:
Fixed bug that when a toast is openend and close it it closes the ShareDialog

Commit:
Merge branch 'feature/PB-118-advanced-sharing' into PB-666-manage-access-requests

CandelR committed:
Merge pull request #732 from internxt/PB-666-manage-access-requests

[PB-666] feature/Manage access requests

Commit:
Added env variable to set env and fix cypress ci tests

CandelR committed:
Merge branch 'master' into feature/PB-118-advanced-sharing

Commit:
Fix NotificationToast error when call dispatch outside the useEffect

aboredvaro committed:
Fixed some styles

Commit:
Changed const in DomainManager

Commit:
Removed duplicated key
